### ISSUE TITLE
[3.3.5] DB/Quest: Protecting Our Own

### ISSUE THREAD

**Issue by Rushor** [TrinityCore/TrinityCore#14736]
TrinityCore rev. https://github.com/TrinityCore/TrinityCore/commit/0148832b322b58a5219ea4393331e1b5937c304c  (3.3.5 branch) (Win64, RelWithDebInfo)

http://wotlk.openwow.com/quest=10457

Bug:
You will get 2 questcredits while using the item=29952 close to the Gos instead of one credit

How to reproduce:
.quest add 
.go object 25442
click the go
you will get 2 credits instead of one

proof:
http://pokit.org/get/?63273bcd6ec86ac70845198558d89cc7.jpg

**Comment by Killyana** [comment]
I'm not sure about this but, for some quests when you get the credit the message: quest completed is shown twice like if the player get the credit twice, maybe it's related.

**Comment by tkrokli** [comment]
> quest completed is shown twice like if the player get the credit twice, maybe it's related.

Yes, I can confirm that what you describe happens on most of the quest related events I have seen. TrinityCore rev. f020f6f 2015-05-16 11:52:01 +0100 (3.3.5 branch) (Win64, Release).

**Comment by Rushor** [comment]
for this quest:

```c++
            case 36030: // Protecting Our Own
                spellInfo->Effects[EFFECT_1].Effect = 0;
                break;
```

**Comment by Aokromes** [comment]
This fix is valid? if yes push it.

**Comment by tkrokli** [comment]
TrinityCore rev. 4bf9f0b73c4a 2016-06-15 13:51:26 +0200 (3.3.5 branch) (Win64, Release) /
TDB 335.61 + updates up to and including `2016_06_15_08_world.sql`

Confirming that this issue still exists, as well as the game object can be used multiple times to complete the quest without searching for other objects of the same kind. Missing condition?

**Comment by illfated** [comment]
Updated quest links, since the old link in the main ticket comment no longer resolves to a host:
- https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?quest=10457/protecting-our-own
- https://www.wowhead.com/quest=10457/protecting-our-own

Does anyone know if the 2015 suggestion from Rushor (https://github.com/TrinityCore/TrinityCore/issues/14736#issuecomment-167243650) is a hack or not?

(copy-paste from the comment:)

```c++
            case 36030: // Protecting Our Own
                spellInfo->Effects[EFFECT_1].Effect = 0;
                break;
```

**Comment by offl** [comment]
```cpp
        case GameObjectActions::Open:
            if (Unit* unitCaster = m_caster->ToUnit())
                gameObjTarget->Use(unitCaster);
```
Less hacky solution is figure out can we make `Activate Object: (Open)` effect not give quest credit
I checked all spells and think it's possible but probably will break some of spells and will require additional fixes to some of 
them to give quest credit properly. For example https://github.com/TrinityCore/TrinityCore/issues/6778#issuecomment-113130752 and https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=55647#used-by-item

**Comment by offl** [comment]
With https://github.com/TrinityCore/TrinityCore/issues/14736#issuecomment-167243650 we'll be not able to script event https://youtu.be/l-f4CatCmjg?t=8
That Open makes object cast spell, once creature is hit, it starts an event
